addons/sale/models/sale.py

```diff
@@ -342,25 +355,28 @@ def onchange_partner_id(self):
             })
             return
 
-        addr = self.partner_id.address_get(['delivery', 'invoice'])
-        partner_user = self.partner_id.user_id or self.partner_id.commercial_partner_id.user_id
+        partner = self.partner_id
+        self = self.with_context(force_company=(self.company_id or self.env.company).id).with_context(lang=partner.lang)
+        addr = partner.address_get(['delivery', 'invoice'])
+        partner_user = partner.user_id or partner.commercial_partner_id.user_id
         values = {
-            'pricelist_id': self.partner_id.property_product_pricelist and self.partner_id.property_product_pricelist.id or False,
-            'payment_term_id': self.partner_id.property_payment_term_id and self.partner_id.property_payment_term_id.id or False,
+            # VFE TODO don't set property_product_pricelist on all partners.
```

Comment:
Remove
* do we still want to set the pricelist magically if not given to create?

Reply:
Yes, to bring back

Reply:
Yes